src/azure-cli/azure/cli/command_modules/appconfig/_snapshot_custom_client.py

```diff
@@ -0,0 +1,540 @@
+            raise ValueError("Retention period value should be a non-negative integer value.")
+
+        request_body["retention_period"] = retention_period
+
+    if tags:
+        request_body["tags"] = tags
+
+    return _build_request(
+        "/snapshots/{name}",
+        RequestMethod.PUT,
+        path_arguments={"name": name},
+        json=request_body,
+        if_match=if_match,
+        if_none_match=if_none_match,
+        sync_token=sync_token,
+        **kwargs
+    )
+
+
+def build_list_snapshot_keys_request(
+    name,
+    select=None,
+    sync_token=None,
+    **kwargs
+):
+    _params = {"snapshot": name}
+
+    if select is not None:
+        _params["$Select"] = Serializer().query("select", select, "[str]", div=",")
+
+    return _build_request(
+        "/kv",
+        RequestMethod.GET,
+        sync_token=sync_token,
+        params=_params,
+        **kwargs
+    )
+
+
+def _convert_request(request, files=None):
+    data = request.content if not files else None
+    request = HttpRequest(method=request.method, url=request.url, headers=request.headers, data=data)
+    if files:
+        request.set_formdata_body(files)
+    return request
+
+
+def _format_url_section(template, **kwargs):
+    components = template.split("/")
+    while components:
+        try:
+            return template.format(**kwargs)
+        except KeyError as key:
+            formatted_components = template.split("/")
+            components = [c for c in formatted_components if "{}".format(key.args[0]) not in c]
+            template = "/".join(components)
+
+
+class AppConfigSnapshotClient:
+
+    def __init__(self, appConfigClient):
+        appConfiguration = appConfigClient._impl
+        self._serializer = appConfiguration._serialize
+        self._deserializer = appConfiguration._deserialize
+        self._sync_token = appConfiguration._config.sync_token
+        self._endpoint = appConfiguration._config.endpoint
+        self._client = appConfiguration._client
+
+    @classmethod
+    def from_connection_string(cls, connection_string):
+        return cls(AzureAppConfigurationClient.from_connection_string(connection_string))
+
+    def begin_create_snapshot(
+            self,
+            name,
+            filters,
+            composition_type=None,
+            retention_period=None,
+            tags=None,
+            if_match=None,
+            if_none_match=None,
+            **kwargs):
+        """
+        Poll after a given interval (default 5s) to ensure that the snapshot is in 'ready' status.
+        The request times out after 30s by default unless specified otherwise.
+        """
+        timeout = kwargs.pop("timeout", 30)
+        polling_interval = kwargs.pop("polling_interval", 5)
+
+        from datetime import datetime
+
+        def _get_elapsed_time(start_time: datetime):
+            return (datetime.now() - start_time).total_seconds()
+
+        print("Starting...")
```

Comment:
This prints to the user. Since the method waits to ensure that the snapshot status is "ready" before completing, the idea is to make sure it does not appear as though the cli is frozen. So there is a "Starting", then "Provisioning" update.